src/libpmem2/map_posix.c

```diff
@@ -299,6 +307,34 @@ pmem2_map(const struct pmem2_config *cfg, struct pmem2_map **map_ptr)
 
 	LOG(3, "mapped at %p", addr);
 
+	bool eADR = (pmem2_auto_flush() == 1);
+	enum pmem2_granularity available_min_granularity =
+		get_min_granularity(eADR, map_sync);
+
+	if (available_min_granularity > cfg->requested_max_granularity) {
+		/* requested CACHE_LINE, available PAGE */
+		if (cfg->requested_max_granularity ==
```

Comment:
Maybe put these into a matrix:
                   1                2              3  (available)
     1           ""                ""              ""   
     2     "ERROR"         ""              ""
     3     "ERROR"    "ERROR"      ""

just a suggestion. This will make sure you don't forget any, and the `if` galore will be simpler.